File: Domain/Products/ProductBlank.cs
namespace Domain.Products
{
    public class ProductBlank
    {
        public Guid? Id { get; set; }
        public String? Name { get; set; }
        public ProductCategory? Category { get; set; }
        public Guid? GroupId { get; set; }
        public String? Description { get; set; }
        public Boolean? IsSale { get; set; }
        public Decimal? Price { get; set; }
        public Boolean? IsRemoved { get; set; }
    }
}

File: WebApp/ClientApp/src/domain/products/models/productBlank.ts
import { Product } from "./product";
import { ProductCategory } from "./productCategory";

export interface ProductBlank{
    id: string | null
    name: string | null
    category: ProductCategory| null
    groupId: string | null
    description: string| null
    issale: boolean
    price: number| null 
}


export namespace ProductBlank{

    export function getDefault(): ProductBlank{
        return {
            id: null,
            name: null,
            category: null,
            description: null,
            issale: false,
            price: null,
            groupId: null
        }
    }

    export function fromProduct(product: Product): ProductBlank{
        return {
            id: product.id,
            name: product.name,
            description: product.description,
            issale: product.issale,
            category: product.category,
            groupId: product.groupid,
            price: product.price
        }
    }
}